add initial unit test

dd-caleb opened the pull request:
This is an initial unit test for the metdata builder. It loads a sample DLL using the hosting API and then runs the `store_wrapper_method_ref ` function.

lucaspimentel commented on test/Datadog.Trace.ClrProfiler.Native.Tests/metadata_builder_tests.cpp:
```diff
@@ -0,0 +1,90 @@
+    HRESULT hr;
+    hr = CLRCreateInstance(CLSID_CLRMetaHost, IID_ICLRMetaHost,
+                           (VOID**)&metahost);
+    ASSERT_FALSE(FAILED(hr));
```
Not a big deal, but there's a `SUCCEEDED(hr)` macro that could make this code a little more readable than these double negatives.

lucaspimentel commented on test/Datadog.Trace.ClrProfiler.Native.Tests/metadata_builder_tests.cpp:
```diff
@@ -0,0 +1,90 @@
+  MetadataBuilder* metadataBuilder_;
+
+  void SetUp() override {
+    ICLRMetaHost* metahost = NULL;
```
Use the standard `nullptr` instead of `NULL`? (I know the current C++ code sets a bad example, and that's my fault, but we should try to stick to modern C++ going forwards.)

lucaspimentel commented on src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj:
```diff
@@ -138,9 +138,10 @@
     <ClInclude Include="integration.h" />
     <ClInclude Include="integration_loader.h" />
     <ClInclude Include="IntegrationType.h" />
+    <ClInclude Include="iterators.h" />
```
It looks like `iterators.h` wasn't added to the repo which is causing the [CI build](https://datadog-apm.visualstudio.com/dd-trace-csharp/default/_build/results?buildId=126&_a=summary&view=logs) to fail.

dd-caleb commented:
@lucaspimentel I added the iterators code. It converts ms-style enums to c++ for/range loops. I also added a second test for a method in a separate DLL.

lucaspimentel commented on src/Datadog.Trace.ClrProfiler.Native/metadata_builder.cpp:
```diff
@@ -24,27 +25,14 @@ HRESULT MetadataBuilder::emit_assembly_ref(
-      assemblyImport->CloseEnum(hEnum);
-      LOG_APPEND(L"Could not find an AssemblyRef to " << assembly_name);
-      return E_FAIL;
+  for (mdAssemblyRef assembly_ref : trace::EnumAssemblyRefs(assemblyImport)) {
```
Nice!
